File: szyfr_cezara_Adam_Zych.cpp
/*
 * szyfr_cezara. cpp
 */


#include <iostream>

using namespace std;

void szyfruj(char tekst[], int klucz){
    klucz = klucz % 26;
    int i = 0;
    while (tekst[i] != '\0') {
		if (tekst[i] == ' ')
			continue;

		if ((int)tekst[i] >= 65 && (int)tekst[i] <= 90)
			tekst[i] = tekst[i] + 32;

		if ((int)tekst[i] + klucz > 122)
			tekst[i] = char((int)tekst[i] - 26 + klucz);
		else
			tekst[i] = char((int)tekst[i] + klucz);
		i++;
	}
	cout << tekst << endl;
}

int main(int argc, char **argv)
{
    char tekst[100];
    int klucz = 3;
    cout << "Podaj tekst: " << endl;
    cin >> tekst;
    cout << "Podaj klucz; ";
    cin >> klucz;
    szyfruj(tekst, klucz);
    cout << tekst;

	return 0;
}

File: szyfr_cezara_Adam_Zych.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def szyfruj_cezar(tekst, klucz):
    klucz = klucz % 26
    szyfrogram = ''
    i = 0
    for znak in tekst:
        if(tekst[i] == ' '):
            szyfrogram += ' '
            continue
        znak = znak.lower()
        if ord(znak) + klucz > 122:
            szyfrogram += chr(ord(znak) + klucz - 26)
        else:
            szyfrogram += chr(ord(znak) + klucz)
    return szyfrogram


def main(args):
    tekst = raw_input("Podaj tekst: ")
    klucz = int(raw_input("Podaj klucz: "))
    szyfrogram = szyfruj_cezar(tekst, klucz)
    print szyfrogram
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
